services/src/main/java/org/jboss/windup/web/services/service/AnalysisContextService.java

```diff
@@ -39,16 +43,35 @@ public AnalysisContext createDefaultAnalysisContext()
     public AnalysisContext create(AnalysisContext analysisContext)
     {
         this.ensureSystemRulesPathsPresent(analysisContext);
+        this.loadPackagesToAnalysisContext(analysisContext);
         entityManager.persist(analysisContext);
+
         return analysisContext;
     }
 
+    protected void loadPackagesToAnalysisContext(AnalysisContext analysisContext)
+    {
+        analysisContext.setIncludePackages(this.loadPackagesFromPersistenceContext(analysisContext.getIncludePackages()));
+        analysisContext.setExcludePackages(this.loadPackagesFromPersistenceContext(analysisContext.getExcludePackages()));
+    }
+
+    protected Set<Package> loadPackagesFromPersistenceContext(Collection<Package> detachedPackages)
+    {
+        return detachedPackages.stream()
+                .filter(aPackage -> aPackage.getId() != null && aPackage.getId() != 0)
+                .map(aPackage -> this.entityManager.find(Package.class, aPackage.getId()))
+                .filter(aPackage -> aPackage != null)
+                .collect(Collectors.toSet());
+    }
```

Comment:
I think that it is reasonable to just ignore it if the user sends one that hasn't been persisted before. I can't think of a reason that the API user should be doing that.